app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,33 @@
+class FriendshipsController < ApplicationController
+  before_action :require_user
+  def index;end
+
+  def create
+    leader = User.find(params[:leader_id])
+    if current_user.id == params[:leader_id].to_i
+      flash["danger"] = "You can't follow yourself"
+      redirect_to user_path(current_user)
+      return
+    elsif current_user.follows?(leader)
+      flash["danger"] = "You are already following #{leader.first_name}"
+      redirect_to user_friendships_path(current_user)
+      return
+    else
+      newfriends = Friendship.new(leader_id: params[:leader_id], follower_id: current_user.id) 
```

Comment:
You don't need to pass in ids always.

```ruby
Friendship.new leader: leader, follower: current_user
```

Also `newfriends` as variable name is strange. Call it either `friendship` as its an instance of Friendship or `relationship`.